index.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+
+const fp = require('fastify-plugin')
+const fs = require('fs')
+const path = require('path')
+const HLRU = require('hashlru')
+const supportedEngines = ['ejs', 'pug', 'handlebars']
+
+function fastifyView (fastify, opts, next) {
+  fastify.decorateReply('view', view)
+
+  if (!opts.engine) {
+    next(new Error('Missing engine'))
+    return
+  }
+
+  const type = Object.keys(opts.engine)[0]
+  if (supportedEngines.indexOf(type) === -1) {
+    next(new Error(`${type} not yet supported, PR? :)`))
+    return
+  }
+
+  const engine = opts.engine[type]
+  const options = opts.options || {}
+  const templatesDir = path.join(process.cwd(), opts.templates || './')
+  const lru = HLRU(opts.maxCache || 100)
+
+  function view (page, data) {
+    if (!page || !data) {
+      this.send(new Error('Missing data'))
+      return
+    }
+
+    const toHtml = lru.get(page)
+
+    if (toHtml) {
+      this.header('Content-Type', 'text/html').send(toHtml(data))
+      return
+    }
+
+    fs.readFile(templatesDir + page, 'utf8', readCallback(this, page, data))
+  }
```

Comment:
Oh you are right :P
So in your opinion a better solution could be something like
```js
const prod = process.env.NODE_ENV === 'production'
...
const toHtml = lru.get(page)

if (toHtml && prod) {
   this.header('Content-Type', 'text/html').send(toHtml(data))
   return
}
```